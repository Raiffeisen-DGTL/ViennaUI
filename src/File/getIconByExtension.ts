import React from 'react';
import {
    XlsIcon,
    JpgIcon,
    JpegIcon,
    DocIcon,
    DocxIcon,
    CameraPhotoIcon,
    PdfIcon,
    ArchiveIcon,
    DocumentIcon,
    Music1Icon,
    Music2Icon,
    MovieIcon,
    SettingsIcon,
    DocEditIcon,
    DocPreviewIcon,
    IconProps,
} from 'vienna.icons';
import { isProperKey } from '../Utils';

const EXTENSION_ICONS = {
    xls: XlsIcon,
    xlsx: XlsIcon,
    jpg: JpgIcon,
    jpeg: JpegIcon,
    jpegx: CameraPhotoIcon,
    png: CameraPhotoIcon,
    doc: DocIcon,
    docx: DocxIcon,
    pdf: PdfIcon,
    zip: ArchiveIcon,
    rar: ArchiveIcon,
    rar5: ArchiveIcon,
    '7z': ArchiveIcon,
    tar: ArchiveIcon,
    gz: ArchiveIcon,
    mp3: Music1Icon,
    wma: Music1Icon,
    wav: Music1Icon,
    flac: Music1Icon,
    midi: Music2Icon,
    ogg: Music2Icon,
    avi: MovieIcon,
    mov: MovieIcon,
    divx: MovieIcon,
    mp4: MovieIcon,
    mpg: MovieIcon,
    wmv: MovieIcon,
    txt: DocEditIcon,
    dll: SettingsIcon,
    deb: SettingsIcon,
    dev: SettingsIcon,
    ppt: DocPreviewIcon,
    pptx: DocPreviewIcon,
};

export const getIconByExtension = (extension: string): React.FC<IconProps> => {
    const key = extension.toLowerCase();

    return isProperKey(EXTENSION_ICONS, key) ? EXTENSION_ICONS[key] : DocumentIcon;
};
