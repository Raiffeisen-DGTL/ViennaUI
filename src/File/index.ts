export { File, type FCCFile, type FileProps } from './File';
