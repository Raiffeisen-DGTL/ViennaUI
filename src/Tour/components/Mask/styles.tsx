import styled, { css } from 'styled-components';
import { color } from 'vienna.tokens';

export const Wrapper = styled.div<{ $isOverlayHidden: boolean }>`
    left: 0;
    top: 0;
    position: fixed;
    z-index: 99999;
    pointer-events: none;
    color: ${color.primary.brand.primary};
    ${({ $isOverlayHidden }) =>
        $isOverlayHidden
            ? css`
                  opacity: 0;
              `
            : css`
                  opacity: 0.24;
              `};
`;

export const SvgWrapper = styled.svg<{
    $windowWidth: number;
    $windowHeight: number;
    $wpt: number | string;
    $wpl: number | string;
}>`
    position: fixed;
    ${({ $windowHeight, $windowWidth, $wpl, $wpt }) => css`
        width: ${$windowWidth}px;
        height: ${$windowHeight}px;
        left: ${Number($wpl)}px;
        top: ${Number($wpt)}px;
    `};
`;

export const MaskArea = styled.rect<{ $x: number; $y: number; $width: number; $height: number; $rx: number }>`
    fill: black;
    ${({ $height, $width, $x, $y, $rx = 16 }) => css`
        x: ${$x}px;
        y: ${$y}px;
        width: ${$width}px;
        height: ${$height}px;
        rx: ${$rx};
    `};
`;

export const MaskRect = styled.rect<{ $windowWidth: number; $windowHeight: number; $maskId?: string }>`
    fill: currentColor;
    ${({ $windowHeight, $windowWidth, $maskId }) => {
        return css`
            x: 0px;
            y: 0px;
            width: ${$windowWidth}px;
            height: ${$windowHeight}px;
            mask: url(#${$maskId});
        `;
    }};
`;

export const HighlightedArea = styled.rect<{
    $x: number;
    $y: number;
    $width: number;
    $height: number;
}>`
    pointer-events: auto;
    fill: transparent;
    display: none;
    ${({ $x, $y, $width, $height }) => css`
        x: ${$x}px;
        y: ${$y}px;
        width: ${$width}px;
        height: ${$height}px;
    `};
`;

export const ClickArea = styled.rect<{
    $width: number;
    $height: number;
    $top: number;
    $left: number;
    $windowWidth: number;
    $windowHeight: number;
    $clipId?: string;
}>`
    fill: currentColor;
    pointer-events: auto;
    ${({ $clipId, $windowHeight, $windowWidth }) => css`
        x: 0px;
        y: 0px;
        width: ${$windowWidth}px;
        height: ${$windowHeight}px;
        clip-path: url(#${$clipId});
    `};
`;
