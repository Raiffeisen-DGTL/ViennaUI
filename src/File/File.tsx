import React, { useCallback, useState, useEffect, PropsWithChildren } from 'react';
import { Cross16Icon } from 'vienna.icons';
import {
    Box,
    Icon,
    Main,
    Data,
    DataDescription,
    DataInfo,
    Name,
    CloseIcon,
    Info,
    Progress,
    ImgIcon,
    PropsBox,
} from './File.styles';
import { Progressbar } from '../Progressbar';
import { getFileExtension, formatDate, formatBytes, ComponentWrapper } from '../Utils';
import { Link } from '../Link';
import { Flex } from '../Flex';
import { useLocalization } from '../Localization';
import { FileLocalizationProps, defaultFileLocalization } from './localization';
import { getIconByExtension } from './getIconByExtension';
import { BoxStyled } from '@/Utils/styled';

export const defaultFileTestId: FileProps['testId'] = {
    icon: 'file_icon',
    name: 'file_name',
    cross: 'file_cross',
};

export interface FCCFile extends Partial<File> {
    id?: string;
    url?: string;
    date?: string | Date;
    progress?: number;
    disabled?: boolean;
    closable?: boolean;
    loading?: boolean;
}

export interface FileProps extends BoxStyled<typeof Box, PropsBox>, FileLocalizationProps {
    /** Файл */
    file: FCCFile;
    altText?: string;
    description?: string;
    invalid?: boolean;
    disabled?: boolean;
    /** Наличие крестика */
    closable?: boolean;
    /** Процент загрузки */
    progress?: number;
    /** Отображение загрузки */
    loading?: boolean;
    /** Отображение произвольного расширения */
    extension?: string;
    /** Наличие ссылки в названии */
    hasNameLink?: boolean;
    /** Показ/скрытие инфо о файле в случае ошибки (по умолчанию выключено) */
    hideAltTextOnError?: boolean;
    onClick?: (file: FCCFile) => void;
    onDelete?: (file: FCCFile) => void;
    confirmDelete?: (file: FCCFile) => boolean | Promise<boolean>;
    /** Включить анимацию. По умолчанию, true */
    animation?: boolean;
    /** включает возможность скачать файл c переданными названием при клике на него */
    download?: string;
    testId?: {
        icon?: string;
        name?: string;
        cross?: string;
    };
}

const getImage = (file: FCCFile) => {
    const type = file.type ?? '';
    if (/image\/.*/.test(type) && !type.includes('tiff')) {
        return file.url;
    }

    return '';
};

const getFileIcon = (file: FCCFile) => {
    const extension = getFileExtension(file);
    return getIconByExtension(extension);
};

export const File: React.FC<PropsWithChildren<FileProps>> = (props) => {
    const {
        file,
        confirmDelete,
        onDelete,
        onClick,
        invalid,
        children,
        closable = true,
        disabled = false,
        progress = 0,
        loading,
        altText,
        description,
        extension,
        hasNameLink,
        localization,
        animation = true,
        hideAltTextOnError = false,
        download,
        testId = defaultFileTestId,
        ...attrs
    } = props;

    const l10n = useLocalization(localization, defaultFileLocalization);

    const [toggle, setToggle] = useState(false);

    const FileIcon = getFileIcon(file);

    const close = useCallback(async () => {
        if (
            typeof confirmDelete === 'undefined' ||
            (typeof confirmDelete === 'function' && (await confirmDelete(file)))
        ) {
            if (!disabled && closable && typeof onDelete === 'function') {
                setToggle(false);
                setTimeout(() => {
                    onDelete(file);
                }, 200);
            }
        }
    }, [file, onDelete, disabled, closable, confirmDelete]);

    const handleClick = useCallback(() => {
        if (!disabled && typeof onClick === 'function') {
            onClick(file);
        }
    }, [file, onClick, disabled]);

    const handleKeyDownName = useCallback(
        async (e: React.KeyboardEvent<HTMLDivElement>) => {
            if (e.key === 'Enter') {
                handleClick();
            }
            if (e.key === 'Delete' || e.key === 'Backspace') {
                await close();
            }
        },
        [close, handleClick]
    );

    const handleKeyDownBtn = useCallback(
        async (e: React.KeyboardEvent<HTMLDivElement>) => {
            if (e.key === 'Enter') {
                await close();
            }
        },
        [close]
    );

    useEffect(() => {
        setToggle(true);
    }, []);

    const constructAdditionalInform = () => {
        const infoData = [file.date && formatDate(file.date), file.size && formatBytes(file.size)];
        const info = infoData.filter(Boolean).join(' • ');
        return info || altText;
    };
    const textInDescription = children ?? description;

    return (
        <Box {...attrs} $toggle={toggle} $animation={animation}>
            <ComponentWrapper
                component={(hasNameLink || download) && file.url && !disabled ? Link : undefined}
                props={{ href: file.url, target: '_blank', download }}>
                <Icon $image={getImage(file)} $disabled={disabled} data-testid={testId?.icon} onClick={handleClick}>
                    <ImgIcon>{extension ?? <FileIcon size='l' />}</ImgIcon>
                </Icon>
            </ComponentWrapper>
            <Main>
                <Info>
                    <ComponentWrapper
                        component={(hasNameLink || download) && file.url && !disabled ? Link : undefined}
                        props={{ href: file.url, target: '_blank', download }}>
                        <Name
                            $hover={!!(hasNameLink || download || onClick)}
                            $disabled={disabled}
                            tabIndex={0}
                            data-testid={testId?.name}
                            onClick={handleClick}
                            onKeyDown={handleKeyDownName}>
                            {file.name}
                        </Name>
                    </ComponentWrapper>
                    <Data>
                        {loading && `${l10n('ds.file.loading')} ${progress}%`}
                        {!loading && (
                            <Flex wrap={'wrap'} gap={'s3'}>
                                {textInDescription && (
                                    <Flex.Item>
                                        <DataDescription $invalid={invalid} $disabled={disabled}>
                                            {textInDescription}
                                        </DataDescription>
                                    </Flex.Item>
                                )}
                                {!hideAltTextOnError && (
                                    <Flex.Item>
                                        <DataInfo $disabled={disabled}>{constructAdditionalInform()}</DataInfo>
                                    </Flex.Item>
                                )}
                            </Flex>
                        )}
                    </Data>
                </Info>
                {Boolean(progress) && loading && (
                    <Progress>
                        <Progressbar size='s' color='oslo100' value={progress} />
                    </Progress>
                )}
            </Main>
            {closable && (
                <CloseIcon
                    $disabled={disabled}
                    tabIndex={0}
                    data-testid={testId?.cross}
                    onClick={close}
                    onKeyDown={handleKeyDownBtn}>
                    <Cross16Icon size={'s'} />
                </CloseIcon>
            )}
        </Box>
    );
};

File.displayName = 'File';
