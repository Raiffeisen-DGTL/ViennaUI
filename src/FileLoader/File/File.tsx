import React, { useCallback, useState, useEffect, PropsWithChildren } from 'react';
import { CloseCancelX } from '@fcc/icons';
import { Box, Icon, Main, Data, Name, CloseIcon, Info, Progress, ImgIcon } from './File.styles';
import { Progressbar } from '../../Progressbar';
import { getFileExtension, formatDate, formatBytes } from '../../Utils';
import { useLocalization } from '../../Localization';
import { FileLoaderLocalizationProps, defaultFileLoaderLocalization } from '../localization';
import { getIconByExtension } from './getIconByExtension';

export interface FCCFile extends Partial<File> {
    id?: string;
    url?: string;
    date?: string | Date;
    progress?: number;
    disabled?: boolean;
    closable?: boolean;
    loading?: boolean;
}

export interface FileProps extends FileLoaderLocalizationProps {
    /** Файл */
    file: FCCFile;
    altText?: string;
    invalid?: boolean;
    disabled?: boolean;
    /** Наличие крестика */
    closable?: boolean;
    /** Процент загрузки */
    progress?: number;
    /** Отображение загрузки */
    loading?: boolean;
    /** Отображение произвольного расширения */
    extension?: string;
    onClick?: (file: FCCFile) => void;
    onDelete?: (file: FCCFile) => void;
    confirmDelete?: (file: FCCFile) => boolean;
}

const getImage = (file: FCCFile) => {
    if (/image\/.*/.test(file.type ?? '')) {
        return file.url;
    }

    return '';
};

const getFileIcon = (file: FCCFile) => {
    const extension = getFileExtension(file);
    return getIconByExtension(extension);
};

export const File: React.FC<PropsWithChildren<FileProps>> = (props) => {
    const {
        file,
        confirmDelete,
        onDelete,
        onClick,
        invalid,
        children,
        closable = true,
        disabled = false,
        progress = 0,
        loading,
        altText,
        extension,
        localization,
    } = props;

    const l10n = useLocalization(localization, defaultFileLoaderLocalization);

    const [toggle, setToggle] = useState(false);

    const FileIcon = getFileIcon(file);

    const close = useCallback(() => {
        if (typeof confirmDelete === 'undefined' || (typeof confirmDelete === 'function' && confirmDelete(file))) {
            if (!disabled && closable && typeof onDelete === 'function') {
                setToggle(false);
                setTimeout(() => {
                    onDelete(file);
                }, 200);
            }
        }
    }, [file, onDelete, disabled, closable, confirmDelete]);

    const handleClick = useCallback(() => {
        if (!disabled && typeof onClick === 'function') {
            onClick(file);
        }
    }, [file, onClick, disabled]);

    const handleKeyDownName = useCallback(
        (e: React.KeyboardEvent<HTMLDivElement>) => {
            if (e.key === 'Enter') {
                handleClick();
            }
            if (e.key === 'Delete' || e.key === 'Backspace') {
                close();
            }
        },
        [close, handleClick]
    );

    const handleKeyDownBtn = useCallback(
        (e: React.KeyboardEvent<HTMLDivElement>) => {
            if (e.key === 'Enter') {
                close();
            }
        },
        [close]
    );

    useEffect(() => {
        setToggle(true);
    }, []);

    const constructAdditionalInform = () => {
        if (file.date && file.size) {
            return `${formatDate(file.date)} • ${formatBytes(file.size)}`;
        }

        return altText;
    };

    return (
        <Box $toggle={toggle}>
            <Icon $image={getImage(file)}>
                <ImgIcon>{extension ?? <FileIcon size='l' />}</ImgIcon>
            </Icon>
            <Main>
                <Info>
                    <Name tabIndex={0} onClick={handleClick} onKeyDown={handleKeyDownName}>
                        {file.name}
                    </Name>
                    <Data $invalid={invalid}>
                        {loading && `${l10n('ds.fileLoader.file.loading')} ${progress}%`}
                        {!loading && (children || constructAdditionalInform())}
                    </Data>
                </Info>
                {Boolean(progress) && loading && (
                    <Progress>
                        <Progressbar size='s' color='oslo100' value={progress} />
                    </Progress>
                )}
            </Main>
            {closable && (
                <CloseIcon tabIndex={0} onClick={close} onKeyDown={handleKeyDownBtn}>
                    <CloseCancelX size={'s'} />
                </CloseIcon>
            )}
        </Box>
    );
};
