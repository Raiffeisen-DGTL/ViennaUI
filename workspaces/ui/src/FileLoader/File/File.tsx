import React, { useCallback, useState, useEffect } from 'react';
import { Close } from 'vienna.icons';
import { Box, Icon, Main, Data, Name, CloseIcon, Info, Progress, Img } from './File.styles';
import { Progressbar } from '../../Progressbar';
import { getFileExtension, formatDate, formatBytes } from '../../Utils';
import { useLocalization } from '../../Localization';
import { FileLoaderLocalizationProps, defaultFileLoaderLocalization } from '../localization';

export interface FCCFile extends Partial<File> {
    id?: string;
    url?: string;
    date: string | Date;
    progress?: number;
    disabled?: boolean;
    closable?: boolean;
    loading?: boolean;
}

export interface Props extends FileLoaderLocalizationProps {
    /** Файл */
    file: FCCFile;
    altText?: string;
    invalid?: boolean;
    /** Строка информации (там где вывод даты) */
    children?: React.ReactNode;
    disabled?: boolean;
    /** Наличие крестика */
    closable?: boolean;
    /** Процент загрузки */
    progress?: number;
    /** Отображение загрузки */
    loading?: boolean;
    /** Отображение произвольного расширения */
    extension?: string;
    onClick?: (file: FCCFile) => void;
    onDelete?: (file: FCCFile) => void;
}

const getImage = (file: FCCFile) => {
    if (/image\/.*/.test(file.type ?? '')) {
        return file.url;
    }

    return '';
};

export const File: React.FC<Props> = (props: Props) => {
    const {
        file,
        onDelete,
        onClick,
        invalid,
        children,
        closable = true,
        disabled = false,
        progress = 0,
        loading,
        altText,
        extension,
    } = props;

    const localization = useLocalization(props, defaultFileLoaderLocalization);

    const [toggle, setToggle] = useState(false);

    const close = useCallback(() => {
        if (!disabled && closable && typeof onDelete === 'function') {
            setToggle(false);
            setTimeout(() => {
                onDelete(file);
            }, 200);
        }
    }, [file, onDelete, disabled, closable]);

    const handleClick = useCallback(() => {
        if (!disabled && typeof onClick === 'function') {
            onClick(file);
        }
    }, [file, onClick, disabled]);

    const handleKeyDownName = useCallback(
        (e: React.KeyboardEvent<HTMLDivElement>) => {
            if (e.key === 'Enter') {
                handleClick();
            }
            if (e.key === 'Delete' || e.key === 'Backspace') {
                close();
            }
        },
        [close, handleClick]
    );

    const handleKeyDownBtn = useCallback(
        (e: React.KeyboardEvent<HTMLDivElement>) => {
            if (e.key === 'Enter') {
                close();
            }
        },
        [close]
    );

    useEffect(() => {
        setToggle(true);
    }, []);

    const constructAddtitionalInform = () => {
        if (file.date && file.size) {
            return `${formatDate(file.date)} • ${formatBytes(file.size)}`;
        }

        return altText;
    };

    return (
        <Box toggle={toggle}>
            <Icon image={getImage(file)}>
                <Img>{extension ?? getFileExtension(file)}</Img>
            </Icon>
            <Main>
                <Info>
                    <Name tabIndex={0} onClick={handleClick} onKeyDown={handleKeyDownName}>
                        {file.name}
                    </Name>
                    <Data invalid={invalid}>
                        {loading && `${localization('ds.fileLoader.file.loading')} ${progress}%`}
                        {!loading && (children || constructAddtitionalInform())}
                    </Data>
                </Info>
                {Boolean(progress) && loading && (
                    <Progress>
                        <Progressbar size='s' color='oslo100' value={progress} />
                    </Progress>
                )}
            </Main>
            {closable && (
                <CloseIcon tabIndex={0} onClick={close} onKeyDown={handleKeyDownBtn}>
                    <Close size={'s'} />
                </CloseIcon>
            )}
        </Box>
    );
};
