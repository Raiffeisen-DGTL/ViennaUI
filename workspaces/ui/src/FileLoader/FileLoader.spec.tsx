import React from 'react';
import { FileLoader } from './index';
import { CloseIcon, Name } from './File/File.styles';
import { Link } from '..';

it('FileLoader /w content', () => {
    const fileloader = snapshot.render(<FileLoader content={'Перетащите файлы'} />);

    expect(fileloader).toMatchSnapshot();
});

it('FileLoader /w content as ReactNode', () => {
    const content = (
        <>
            Перетащите файлы сюда или{'\u00A0'}
            <Link design='accent'>загрузите документы</Link>
        </>
    );
    const fileloader = snapshot.render(<FileLoader content={content} />);

    expect(fileloader).toMatchSnapshot();
});

it('FileLoader /w content and subcontent', () => {
    const fileloader = snapshot.render(
        <FileLoader content={'Перетащите файлы'} subContent={'JPG, GIF, PNG, PDF или ZIP, до 150kb'} />
    );

    expect(fileloader).toMatchSnapshot();
});

it('FileLoader /w with files', () => {
    const fileloader = snapshot.render(
        <FileLoader content={'Перетащите файлы'}>
            <FileLoader.File file={{ url: '1', date: '', size: 2048, name: 'file 1.doc' }} />
            <FileLoader.File file={{ url: '1', date: '', size: 2048, name: 'file 1.doc' }} />
            <FileLoader.File file={{ url: '1', date: '', size: 2048, name: 'file 1.doc' }} />
        </FileLoader>
    );

    expect(fileloader).toMatchSnapshot();
});

it('FileLoader /w file', () => {
    const file = snapshot.render(<FileLoader.File file={{ url: '1', date: '', size: 2048, name: 'file 1.doc' }} />);

    expect(file).toMatchSnapshot();
});

it('FileLoader /w file progress', () => {
    const file = snapshot.render(
        <FileLoader.File progress={50} loading file={{ url: '1', date: '', size: 2048, name: 'file 1.doc' }} />
    );

    expect(file).toMatchSnapshot();
});

it('FileLoader /w file closable = false', () => {
    const file = snapshot.render(
        <FileLoader.File closable={false} file={{ url: '1', date: '', size: 2048, name: 'file 1.doc' }} />
    );

    expect(file).toMatchSnapshot();
});

it('FileLoader /w file onDelete()', () => {
    const file = snapshot.shallow(<FileLoader.File file={{ url: '1', date: '', size: 2048, name: 'file 1.doc' }} />);

    file.find(CloseIcon).simulate('click');

    expect(file.props().toggle).toEqual(false);
});

it('FileLoader /w file onClick()', () => {
    const fn = jest.fn();
    const file = snapshot.mount(
        <FileLoader.File file={{ url: '1', date: '', size: 2048, name: 'file 1.doc' }} onClick={fn} />
    );
    file.find(Name).simulate('click');

    expect(fn).toBeCalled();
});

it('FileLoader /w file onDelete() keyboard', () => {
    const file = snapshot.shallow(<FileLoader.File file={{ url: '1', date: '', size: 2048, name: 'file 1.doc' }} />);

    file.find(Name).simulate('keydown', { key: 'Delete' });

    expect(file.props().toggle).toEqual(false);
});

it('FileLoader /w file onClick() keyboard', () => {
    const fn = jest.fn();
    const file = snapshot.mount(
        <FileLoader.File file={{ url: '1', date: '', size: 2048, name: 'file 1.doc' }} onClick={fn} />
    );
    file.find(Name).simulate('keydown', { key: 'Enter' });

    expect(fn).toBeCalled();
});
