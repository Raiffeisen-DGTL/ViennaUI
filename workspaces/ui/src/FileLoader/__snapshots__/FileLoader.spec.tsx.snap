// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileLoader /w content 1`] = `
<div
  class="FileLoaderstyles__Wrapper-ymi30o-4 kZTuzU"
>
  <div
    class="FileLoaderstyles__Box-ymi30o-2 eYElVZ"
  >
    <div
      class="FileLoaderstyles__Content-ymi30o-0 igzYZa"
    >
      Перетащите файлы
    </div>
    <input
      accept=""
      class="FileLoaderstyles__InputFile-ymi30o-5 kIScVj"
      multiple=""
      type="file"
      value=""
    />
  </div>
</div>
`;

exports[`FileLoader /w content and subcontent 1`] = `
<div
  class="FileLoaderstyles__Wrapper-ymi30o-4 kZTuzU"
>
  <div
    class="FileLoaderstyles__Box-ymi30o-2 eYElVZ"
  >
    <div
      class="FileLoaderstyles__Content-ymi30o-0 igzYZa"
    >
      Перетащите файлы
    </div>
    <div
      class="FileLoaderstyles__SubContent-ymi30o-1 fMNhrg"
    >
      JPG, GIF, PNG, PDF или ZIP, до 150kb
    </div>
    <input
      accept=""
      class="FileLoaderstyles__InputFile-ymi30o-5 kIScVj"
      multiple=""
      type="file"
      value=""
    />
  </div>
</div>
`;

exports[`FileLoader /w content as ReactNode 1`] = `
<div
  class="FileLoaderstyles__Wrapper-ymi30o-4 kZTuzU"
>
  <div
    class="FileLoaderstyles__Box-ymi30o-2 eYElVZ"
  >
    <div
      class="FileLoaderstyles__Content-ymi30o-0 igzYZa"
    >
      Перетащите файлы сюда или 
      <a
        class="Linkstyles__Box-sc-40rkfc-1 gtrUrS"
      >
        <span
          class="Linkstyles__Wrapper-sc-40rkfc-0 jXPYyj"
        >
          загрузите документы
        </span>
      </a>
    </div>
    <input
      accept=""
      class="FileLoaderstyles__InputFile-ymi30o-5 kIScVj"
      multiple=""
      type="file"
      value=""
    />
  </div>
</div>
`;

exports[`FileLoader /w file 1`] = `
<div
  class="Filestyles__Box-sc-1sbri8o-0 gRRDvd"
>
  <div
    class="Filestyles__Icon-sc-1sbri8o-2 jycNGT"
  >
    <div
      class="Filestyles__Img-sc-1sbri8o-1 ejlgtk"
    >
      doc
    </div>
  </div>
  <div
    class="Filestyles__Main-sc-1sbri8o-3 iiJeMn"
  >
    <div
      class="Filestyles__Info-sc-1sbri8o-7 cvvNqD"
    >
      <div
        class="Filestyles__Name-sc-1sbri8o-4 fSKaNn"
        tabindex="0"
      >
        file 1.doc
      </div>
      <div
        class="Filestyles__Data-sc-1sbri8o-5 jnhkhy"
      />
    </div>
  </div>
  <div
    class="Filestyles__CloseIcon-sc-1sbri8o-6 fjZhUW"
    tabindex="0"
  >
    <svg
      fill="currentColor"
      height="16"
      viewBox="0 0 24 24"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <defs />
      <path
        d="M12 13.414l-7.293 7.293-1.414-1.414L10.586 12 3.293 4.707l1.414-1.414L12 10.586l7.293-7.293 1.414 1.414L13.414 12l7.293 7.293-1.414 1.414L12 13.414z"
      />
    </svg>
  </div>
</div>
`;

exports[`FileLoader /w file closable = false 1`] = `
<div
  class="Filestyles__Box-sc-1sbri8o-0 gRRDvd"
>
  <div
    class="Filestyles__Icon-sc-1sbri8o-2 jycNGT"
  >
    <div
      class="Filestyles__Img-sc-1sbri8o-1 ejlgtk"
    >
      doc
    </div>
  </div>
  <div
    class="Filestyles__Main-sc-1sbri8o-3 iiJeMn"
  >
    <div
      class="Filestyles__Info-sc-1sbri8o-7 cvvNqD"
    >
      <div
        class="Filestyles__Name-sc-1sbri8o-4 fSKaNn"
        tabindex="0"
      >
        file 1.doc
      </div>
      <div
        class="Filestyles__Data-sc-1sbri8o-5 jnhkhy"
      />
    </div>
  </div>
</div>
`;

exports[`FileLoader /w file progress 1`] = `
<div
  class="Filestyles__Box-sc-1sbri8o-0 gRRDvd"
>
  <div
    class="Filestyles__Icon-sc-1sbri8o-2 jycNGT"
  >
    <div
      class="Filestyles__Img-sc-1sbri8o-1 ejlgtk"
    >
      doc
    </div>
  </div>
  <div
    class="Filestyles__Main-sc-1sbri8o-3 iiJeMn"
  >
    <div
      class="Filestyles__Info-sc-1sbri8o-7 cvvNqD"
    >
      <div
        class="Filestyles__Name-sc-1sbri8o-4 fSKaNn"
        tabindex="0"
      >
        file 1.doc
      </div>
      <div
        class="Filestyles__Data-sc-1sbri8o-5 jnhkhy"
      >
        Загружено 50%
      </div>
    </div>
    <div
      class="Filestyles__Progress-sc-1sbri8o-8 exFAuN"
    >
      <div
        class="Progressbarstyles__Box-sc-9u8n53-0 ebQjGZ"
      >
        <div
          class="Progressbarstyles__Line-sc-9u8n53-1 lfqWLM"
        />
        <div
          class="Progressbarstyles__Progress-sc-9u8n53-2 jGYoUl"
          color="oslo100"
          width="50"
        />
      </div>
    </div>
  </div>
  <div
    class="Filestyles__CloseIcon-sc-1sbri8o-6 fjZhUW"
    tabindex="0"
  >
    <svg
      fill="currentColor"
      height="16"
      viewBox="0 0 24 24"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <defs />
      <path
        d="M12 13.414l-7.293 7.293-1.414-1.414L10.586 12 3.293 4.707l1.414-1.414L12 10.586l7.293-7.293 1.414 1.414L13.414 12l7.293 7.293-1.414 1.414L12 13.414z"
      />
    </svg>
  </div>
</div>
`;

exports[`FileLoader /w with files 1`] = `
<div
  class="FileLoaderstyles__Wrapper-ymi30o-4 kZTuzU"
>
  <div
    class="FileLoaderstyles__Box-ymi30o-2 eYElVZ"
  >
    <div
      class="FileLoaderstyles__Content-ymi30o-0 igzYZa"
    >
      Перетащите файлы
    </div>
    <input
      accept=""
      class="FileLoaderstyles__InputFile-ymi30o-5 kIScVj"
      multiple=""
      type="file"
      value=""
    />
  </div>
  <div
    class="FileLoaderstyles__List-ymi30o-3 lmKLMJ"
  >
    <div
      class="Filestyles__Box-sc-1sbri8o-0 gRRDvd"
    >
      <div
        class="Filestyles__Icon-sc-1sbri8o-2 jycNGT"
      >
        <div
          class="Filestyles__Img-sc-1sbri8o-1 ejlgtk"
        >
          doc
        </div>
      </div>
      <div
        class="Filestyles__Main-sc-1sbri8o-3 iiJeMn"
      >
        <div
          class="Filestyles__Info-sc-1sbri8o-7 cvvNqD"
        >
          <div
            class="Filestyles__Name-sc-1sbri8o-4 fSKaNn"
            tabindex="0"
          >
            file 1.doc
          </div>
          <div
            class="Filestyles__Data-sc-1sbri8o-5 jnhkhy"
          />
        </div>
      </div>
      <div
        class="Filestyles__CloseIcon-sc-1sbri8o-6 fjZhUW"
        tabindex="0"
      >
        <svg
          fill="currentColor"
          height="16"
          viewBox="0 0 24 24"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <defs />
          <path
            d="M12 13.414l-7.293 7.293-1.414-1.414L10.586 12 3.293 4.707l1.414-1.414L12 10.586l7.293-7.293 1.414 1.414L13.414 12l7.293 7.293-1.414 1.414L12 13.414z"
          />
        </svg>
      </div>
    </div>
    <div
      class="Filestyles__Box-sc-1sbri8o-0 gRRDvd"
    >
      <div
        class="Filestyles__Icon-sc-1sbri8o-2 jycNGT"
      >
        <div
          class="Filestyles__Img-sc-1sbri8o-1 ejlgtk"
        >
          doc
        </div>
      </div>
      <div
        class="Filestyles__Main-sc-1sbri8o-3 iiJeMn"
      >
        <div
          class="Filestyles__Info-sc-1sbri8o-7 cvvNqD"
        >
          <div
            class="Filestyles__Name-sc-1sbri8o-4 fSKaNn"
            tabindex="0"
          >
            file 1.doc
          </div>
          <div
            class="Filestyles__Data-sc-1sbri8o-5 jnhkhy"
          />
        </div>
      </div>
      <div
        class="Filestyles__CloseIcon-sc-1sbri8o-6 fjZhUW"
        tabindex="0"
      >
        <svg
          fill="currentColor"
          height="16"
          viewBox="0 0 24 24"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <defs />
          <path
            d="M12 13.414l-7.293 7.293-1.414-1.414L10.586 12 3.293 4.707l1.414-1.414L12 10.586l7.293-7.293 1.414 1.414L13.414 12l7.293 7.293-1.414 1.414L12 13.414z"
          />
        </svg>
      </div>
    </div>
    <div
      class="Filestyles__Box-sc-1sbri8o-0 gRRDvd"
    >
      <div
        class="Filestyles__Icon-sc-1sbri8o-2 jycNGT"
      >
        <div
          class="Filestyles__Img-sc-1sbri8o-1 ejlgtk"
        >
          doc
        </div>
      </div>
      <div
        class="Filestyles__Main-sc-1sbri8o-3 iiJeMn"
      >
        <div
          class="Filestyles__Info-sc-1sbri8o-7 cvvNqD"
        >
          <div
            class="Filestyles__Name-sc-1sbri8o-4 fSKaNn"
            tabindex="0"
          >
            file 1.doc
          </div>
          <div
            class="Filestyles__Data-sc-1sbri8o-5 jnhkhy"
          />
        </div>
      </div>
      <div
        class="Filestyles__CloseIcon-sc-1sbri8o-6 fjZhUW"
        tabindex="0"
      >
        <svg
          fill="currentColor"
          height="16"
          viewBox="0 0 24 24"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <defs />
          <path
            d="M12 13.414l-7.293 7.293-1.414-1.414L10.586 12 3.293 4.707l1.414-1.414L12 10.586l7.293-7.293 1.414 1.414L13.414 12l7.293 7.293-1.414 1.414L12 13.414z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
